src/cargo_deny_list.rs

```diff
@@ -0,0 +1,246 @@
+#[cfg_attr(test, derive(Default))]
+struct LicenseList {
+    /// List of licenses used by the dependency.
+    licenses: Vec<String>,
+}
+
+/// Represents the name of a dependency and its URL.
+///
+/// The format of this string is expected to be:
+///
+/// ```text
+/// <CRATE_NAME> <URL>
+/// ```
+///
+/// The URL is formatted as a link to the dependency's page.
+#[derive(Debug, PartialEq, Eq, Hash, Serialize)]
+struct NameAndUrl(String);
+
+impl<'de> Deserialize<'de> for NameAndUrl {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let key = String::deserialize(deserializer)?;
+        let key_split = key.split_whitespace().collect::<Vec<_>>();
+
+        match key_split[..] {
+            [crate_name, _crate_version, url] => Ok(NameAndUrl(format!(
+                "{} {}",
+                crate_name,
+                process_name_with_url(crate_name, url)
+            ))),
+            _ => Err(serde::de::Error::custom("Invalid dependency key format")),
+        }
+    }
+}
+
+/// Given a dependency name and a URL, returns a string with the URL formatted as a link to the
+/// dependency's page.
+///
+/// # Examples
+///
+/// ```ignore
+/// let dep_name = "adler2";
+/// let url_str = "registry+https://github.com/rust-lang/crates.io-index";
+/// let result = process_name_with_url(dep_name, url_str);
+/// assert_eq!(result, "https://crates.io/crates/adler2");
+/// ```
+///
+/// If the URL is from git, the `.git` suffix is removed:
+///
+/// ```ignore
+/// let dep_name = "rust-licenses-noticer";
+/// let url_str = "git+https://github.com/newrelic/rust-licenses-noticer.git";
+/// let result = process_name_with_url(dep_name, url_str);
+/// assert_eq!(result, "https://github.com/newrelic/rust-licenses-noticer");
+/// ```
+fn process_name_with_url(dep_name: &str, url_str: &str) -> String {
+    match url_str {
+        // crates.io
+        "registry+https://github.com/rust-lang/crates.io-index" => {
+            format!("https://crates.io/crates/{}", dep_name)
+        }
+        // remove any prefix for other registries, up until the + symbol
+        s if s.starts_with("git+") => remove_source_prefix(s).trim_end_matches(".git").to_string(),
+        s => remove_source_prefix(s),
+    }
+}
+
+/// Removes the prefix of a URL string up until the + symbol.
+/// If the URL doesn't have a + symbol, the string is returned as is.
+///
+/// # Examples
+///
+/// ```ignore
+/// let url_str = "git+https://github.com/newrelic/rust-licenses-noticer.git";
+/// let result = remove_source_prefix(url_str);
+/// assert_eq!(result, "https://github.com/newrelic/rust-licenses-noticer.git");
+/// ```
+fn remove_source_prefix(url_str: &str) -> String {
+    // Is there a registry indicator? We know it's a registry if it has a + symbol at the end.
+    let sum_symbol_index = url_str.find('+').map(|i| i + 1).unwrap_or(0);
+    url_str.chars().skip(sum_symbol_index).collect::<String>()
+}
```

Comment:
This was originally done in the template itself though only for crates.io references (not git repos). It reduces complexity for the templates (as I find them harder to test), but performs the changes via code logic so it's less flexible for other use cases. Let me know what you prefer.